src/libraries/System.Text.Json/Common/ReflectionExtensions.cs

```diff
@@ -49,9 +49,6 @@ internal static partial class ReflectionExtensions
             Debug.Assert(!baseType.IsInterface);
             Debug.Assert(baseType == baseType.GetGenericTypeDefinition());
 
-            // Work around not being able to use typeof(object) directly during compile-time src gen type analysis.
-            objectType ??= typeof(object);
```

Comment:
If this isn't needed, can the method be changed to not even take an objectType?

Reply:
Done